e2e-tests/src/test/electing_validators.rs

```diff
@@ -35,16 +39,29 @@ fn assert_validators_are_elected_stakers<C: AnyConnection>(
     );
 }
 
+// There are v non-reserved validators and s non-reserved seats. We will have seen all
+// the non-reserved validators after ceil(v / s).
+fn min_num_sessions_to_see_all_non_reserved_validators(
+    non_reserved_count: u32,
+    non_reserved_seats: u32,
+) -> u32 {
+    // Matching done to emphasize handling of `non_reserved_seats` = 0.
+    match non_reserved_seats {
+        0 => 0,
+        _ => (non_reserved_count as f64 / non_reserved_seats as f64).ceil() as u32,
```

Comment:
`(v + s - 1) / s` won't work? or some saturating division instead of `/`

Reply:
Clever, I like it :)

Reply:
I have addressed all of the above :)